astroplan/core.py

```diff
@@ -0,0 +1,275 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+#from ..extern import six
+
+import sys
+#from math import sqrt, pi, exp, log, floor
+#from abc import ABCMeta, abstractmethod
+
+import numpy as np
+
+# Originally authored by Jazmin Berlanga Medina (jazmin.berlanga@gmail.com),
+# and modified by Adrian Price-Whelan (email) and Erik Tollerud (email).
+
+__all__ = ["Observer", "Target", "FixedTarget", "NonFixedTarget",
+           "Constraints", "Observation"]
+
+#__doctest_requires__ = {'*': ['scipy.integrate']}
+
+
+class Observer(object):
+    """
+    Some comments.
+    """
+
+    def __init__(self, name, longitude, latitude, elevation, pressure,
+                 rel_humidity, temp, timezone, description=None):
+        """
+        Comments.
+        """
+        raise NotImplementedError()
+
+    def set_environment():
```

Comment:
You should put some arguments in to all functions that reflect the API in #4. At minimum, the first argument in each method must be `self` (well, kinda, [see here](https://docs.python.org/2/tutorial/classes.html)).

But then, let's look at `set_environment` as an example. In the API, line 96 is:

```python
obs.set_environment(pressure=0.600 * u.bar, relative_humidity=0.2,
                    temperature=10 * u.deg_C)
```

At minimum, we will want to add these arguments in to the function definition. So, `def set_environment()` would become

```python
def set_environment(self, pressure, relative_humidity, temperature):
```

Now, to help in auto-generating the documentation, we'll also want to document what each argument represents and what kind of object we expect. To do this, the convention in Astropy is to follow the [numpydoc styling of docstrings](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#docstring-standard). If you look at the class and method definitions in some astropy module, e.g., [here](https://github.com/astropy/astropy/blob/master/astropy/coordinates/earth.py) you can see an example of this formatting. But the basic idea is to do a brief summary of the functionality, a longer explanation if needed, then describe the input arguments / parameters , then describe what the method or function returns. So, for `set_environment` this might look something like:

```python
def set_environment(self, pressure, relative_humidity, temperature):
    """
    <brief description>
   
    <longer description>

    Parameters
    ----------
    pressure : `~astropy.units.Quantity`
        The pressure at the observatory.
    ...etc.

```

In this case, the method doesn't return anything. Try to do this for each of the functions below, and of course let us know if you get stuck!